projects/distributed-tracing/src/pages/trace-detail/trace-detail.service.ts

```diff
@@ -90,6 +105,7 @@ export class TraceDetailService implements OnDestroy {
         this.specificationBuilder.attributeSpecificationForKey('serviceName'),
         this.specificationBuilder.attributeSpecificationForKey('displaySpanName')
       ],
+      spansTimeRange: this.buildGraphqlTimeRange(timeRange),
```

Comment:
I think we could push this logic down in a couple ways:
- Use the same time range for both, and generate the span time range here, since the broader span time range should also be satisfactory for the trace (currently, it looks like we're not bounding the span request by the time stamp)
- Make the trace gql request only accept a timestamp, not a time range - move all time range generation logic inside the query handler and force all callers to provide a timestamp when querying a single trace.

Reply:
I don't think #1 would work. Trace Id can lie outside of the TR selected at the top. This TR is same as the span time range. So it is not really a broader time.

#2 seems a change with large diff. It scares me at this point.

Reply:
for 1, I think there was a miscommunication - I meant ignore the global TR, and given a timestamp, generate a window like `[-30m, +30m]` or so, and pass that in as the one time range for the request. Because we're filtering by IDs here, being extra generous on that window should be absolutely fine, even though not required. The behavior of the timestamps: 
- A trace timestamp should be equal to the earliest included span timestamp
- (assumption) A trace must complete in 30 minutes (I was curious what the platform assumption here is - it's configurable but defaults to 30s, so I think this is plenty of wiggle room)
- All spans belonging to a trace will have a timestamp within the window defined in the point above

2 is an extension of 1, but I think significantly less code since that window generation that is duplicated everywhere (in your PR as well as my proposal) is all left in one spot. It also has the added benefit of type safety - we're forcing the calls for a single trace to provide a timestamp and we can be assured b the compiler we didn't miss a spot.

Reply:
Yeah this makes sense. Should we do it separately or part of this PR only? This new change will likely affect other places as well. I have to focus on some other priority items as of now. So either we defer this api change for later or we defer this PR till that new commit is in.

Reply:
I have addressed your other comments.

Reply:
I think we should at a minimum do 1 as part of this PR (in other words, revert the separate span time range and update the calculations you've added so they're suitable for the request's "primary" time range) - that will be less code then the current diff and limits the tech debt by keeping the trace request API the same as it currently is. If you think changing the api to take a timestamp instead is a significantly larger diff/amount of work (it doesn't seem to me that it would be, but haven't looked at this code in a bit), that's fine to defer.

Reply:
Then I will continue this next week. Have few other things to take care of before it.